modules/user-access/pages/query-privilege-migration.adoc

```diff
@@ -0,0 +1,136 @@
+====
+
+[#_migrate_rbac_write_query]
+=== Migrate RBAC WRITE_QUERY privilege
+In our previous versions, `WRITE_QUERY` privilege means we make all changes for a query. Hence, `WRITE_QUERY` privilege should be migrated to all privileges except `OWNERSHIP` for all existing queries, including `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` on each query object and `CREATE_QUERY` on the same scope for `WRITE_QUERY` privilege.
+
+Similarly, this migration process should be different for our existing `WRITE_QUERY` privileges when your users are granted with built-in roles or user-defined roles.
+
+==== Migrate RBAC WRITE_QUERY privilege for Built-in roles
+The `WRITE_QUERY` privilege for built-in roles should not be kept since 4.1.0. For the detailed changes of built-in roles, please refer to the table above.
+
+If users were granted built-in roles with `WRITE_QUERY` privileges before upgrade, following privileges, `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY`, should be granted to the users directly with all existing queries in the same scope.
+
+====
+.For example:
+* User `Alice` is granted with the built-in role `designer` in graph `recommend` in 3.10.0, and there are two queries `query1` and `query2` in graph `recommend`. After the upgrade process for 4.1.0, user `Alice` should hold the `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` privileges on queries `query1` and `query2` in graph `recommend`. At the same time, user `Alice` can still create queries in graph `recommend` since she still holds the role `designer` in graph `recommend`, which provide the `CREATE_QUERY` privilege in graph `recommend`.
+====
+
+==== Migrate RBAC WRITE_QUERY privilege for User-Defined roles
+The `WRITE_QUERY` privilege in global and graph level scopes for user-defined roles should not be kept since 4.1.0.
+
+If users were granted user-defined roles with `WRITE_QUERY` privileges before upgrade, following privileges, `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY`, should be granted to the same role with all existing queries in the same scope. At the same time, `CREATE_QUERY` privilege on the same scope should be granted to the same role in the same scope.
+
+====
+.For example:
+* User `Bob` is granted with the user-defined role `myRole` with `WRITE_QUERY` in graph `recommend` in 3.10.0, and there are two queries, `query1` and `query2` in graph `recommend`. After the upgrade process for 4.1.0, role `myRole` should have the `CREATE_QUERY` privilege in graph `recommend`, and `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` privileges on query `query1` and `query2` in graph `recommend`. User `Bob` can still create queries in graph `recommend`, and read, update, drop, install and execute existing queries `query1` and `query2` since he is still granted with role `myRole`.
```

Comment:
```suggestion
==== Migrate RBAC WRITE_QUERY privilege for User-Defined roles
The `WRITE_QUERY` privilege in global and graph level scopes for user-defined roles is deprecated in TigerGraph 4.1.

If users were granted user-defined roles with `WRITE_QUERY` privileges before upgrade, the following privileges `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY`, will all be granted to the same role with all existing queries in the same scope. At the same time, `CREATE_QUERY` privilege on the same scope will be granted to the same role in the same scope.

====
.For example:
* User `Bob` was granted with the user-defined role `myRole` with `WRITE_QUERY` in graph `recommend` in TigerGraph 3.10, and there are two queries, `query1` and `query2` in graph `recommend`. After the upgrade to TigerGraph 4.1, role `myRole` will have the `CREATE_QUERY` privilege in graph `recommend`, and `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` privileges on query `query1` and `query2` in graph `recommend`. User `Bob` can still create queries in graph `recommend`, and read, update, drop, install and execute existing queries `query1` and `query2` since he is still granted the role `myRole`.
```